### ISSUE TITLE
docs: improve jwt docs on passed headers

### ISSUE THREAD

**Issue by papacarp** [hasura/graphql-engine#4895]
I have a Cognito authentication with JWT.   1.2.0 will properly parse the token and use the 'samplersadmin' role.  1.2.1 will not and will set the role to 'x-hasura-default-role'

Example JWT
```
{
  "sub": "ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa",
  "email_verified": false,
  "https://hasura.io/jwt/claims": "{\"x-hasura-allowed-roles\":[\"public\",\"samplersrep\",\"samplersuser\",\"samplersadmin\",\"admin\"],\"x-hasura-default-role\":\"public\",\"x-hasura-user-id\":\"ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa\",\"x-hasura-role\":\"samplersadmin\",\"x-hasura-custom\":\"custom-value\"}",
  "iss": "https://cognito-idp.us-west-2.amazonaws.com/us-west-2_xxxx",
  "phone_number_verified": true,
  "cognito:username": "xxx",
  "aud": "58i571ubbor81mhm9lbe83uegs",
  "event_id": "30f3f1f6-xxxx-xxxx-xxxx-efd99cdce539",
  "token_use": "id",
  "auth_time": 1590105217,
  "phone_number": "+xxx",
  "exp": 1590565179,
  "iat": 1590561579,
  "email": "xxx"
} 
```

## Docker entry and logs for 1.2.0 - OK:
```
 graphql-engine:
    image: hasura/graphql-engine:v1.2.0
```

v1.2.0 Hasura log file entry: *"user_vars":{"x-hasura-role":"samplersadmin"*
```
{"type":"query-log","timestamp":"2020-05-27T07:24:06.707+0000","level":"info","detail":{"request_id":"072bd0e7-9f66-497d-a6b8-23f27845cc3c","generated_sql":null,"query":{"variables":{"userId":"ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa"},"operationName":"ChatEmitOnline","query":"mutation ChatEmitOnline($userId: uuid!) {\n  update_users(_set: {last_seen: \"now()\"}, where: {user_id: {_eq: $userId}}) {\n    affected_rows\n    __typename\n  }\n}\n"}}}
{"type":"websocket-log","timestamp":"2020-05-27T07:24:06.707+0000","level":"info","detail":{"event":{"type":"operation","detail":{"request_id":"072bd0e7-9f66-497d-a6b8-23f27845cc3c","operation_id":"27","query":null,"operation_type":{"type":"completed"},"operation_name":"ChatEmitOnline"}},"connection_info":{"websocket_id":"b050d4e2-41ba-4f37-a684-8968c1c97179","token_expiry":"2020-05-27T07:39:39Z","msg":null},"user_vars":{"x-hasura-role":"samplersadmin","x-hasura-custom":"custom-value","x-hasura-user-id":"ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa"}}}

```

## Docker entry and logs for 1.2.1 - FAIL:
```
 graphql-engine:
    image: hasura/graphql-engine:v1.2.1
```

v1.2.1 Hasura log file entry: *"user_vars":{"x-hasura-role":"public"*
```

{"type":"websocket-log","timestamp":"2020-05-27T07:26:57.489+0000","level":"error","detail":{"event":{"type":"operation","detail":{"request_id":"599452a4-5888-4e24-9657-30a21b7a5d42","operation_id":"44","query":{"variables":{"userId":"ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa"},"operationName":"ChatEmitOnline","query":"mutation ChatEmitOnline($userId: uuid!) {\n  update_users(_set: {last_seen: \"now()\"}, where: {user_id: {_eq: $userId}}) {\n    affected_rows\n    __typename\n  }\n}\n"},"operation_type":{"type":"query_err","detail":{"path":"$","error":"no mutations exist","code":"validation-failed"}},"operation_name":"ChatEmitOnline"}},"connection_info":{"websocket_id":"0855af7d-16a7-4ac2-8bb4-7fced5a9d728","token_expiry":"2020-05-27T07:39:39Z","msg":null},"user_vars":{"x-hasura-role":"public","x-hasura-custom":"custom-value","x-hasura-user-id":"ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa"}}}
{"type":"websocket-log","timestamp":"2020-05-27T07:26:57.489+0000","level":"info","detail":{"event":{"type":"operation","detail":{"request_id":null,"operation_id":"44","query":null,"operation_type":{"type":"completed"},"operation_name":"ChatEmitOnline"}},"connection_info":{"websocket_id":"0855af7d-16a7-4ac2-8bb4-7fced5a9d728","token_expiry":"2020-05-27T07:39:39Z","msg":null},"user_vars":{"x-hasura-role":"public","x-hasura-custom":"custom-value","x-hasura-user-id":"ceb27279-xxxx-xxxx-xxxx-95e8bee1d6aa"}}}

```

Same token is used in both cases.  JWT looks good.  Seems something changed in 1.2.1 that broke this?

**Comment by tirumaraiselvan** [comment]
The spec is given here: https://hasura.io/docs/1.0/graphql/manual/auth/authentication/jwt.html#the-spec

`x-hasura-role` is not used from JWT. If you want the role to come from the JWT, then `x-hasura-default-role` should be given. `x-hasura-role` can be given from the client side, which would be used only if it is present in `x-hasura-allowed-roles`. The token might have worked in v1.2.0 because of a bug related to this which was fixed in v1.2.1.

**Comment by papacarp** [comment]
Ok, I'm happy to have a supported solution to the issue and will implement that way.  However, the documentation should be cleaned up if this is how you expect it to operate:

> A x-hasura-default-role field : indicating the default role of that user i.e. the role that will be used in case x-hasura-role header is not passed.

and then also:

> The default role can be overridden by the x-hasura-role header, while making a request.

I'll change the title of this issue and let you guys close it out if you want.
